proposals/lip-0052.md

```diff
@@ -1439,73 +1560,33 @@ This function returns the `collectionID` of an NFT.
 
 ```python
 def getCollectionID(nftID: NFTID) -> CollectionID:
-
-    if NFTStore[nftID] does not exist:
-        raise Exception("NFT substore entry does not exist")
-
-    return nftID[`LENGTH_CHAIN_ID`:(`LENGTH_CHAIN_ID` + `LENGTH_COLLECTION_ID`)]
-```
-
-#### getAttributesArray
-
-This function returns the `attributesArray` of an NFT.
-
-```python
-def getAttributesArray(nftID: NFTID) -> list[dict[Module, bytes]]:
-
-    if NFTStore[nftID] does not exist:
-        raise Exception("NFT substore entry does not exist")
-
-    return NFTStore[nftID].attributesArray
+    return nftID[LENGTH_CHAIN_ID:(LENGTH_CHAIN_ID + LENGTH_COLLECTION_ID)]
 ```
 
-#### getAttributes
-
-This function returns the attributes of an NFT set by a specific module.
-
-```python
-def getAttributes(
-    module: Module,
-    nftID: NFTID
-) -> bytes:
-
-    if NFTStore[nftID] does not exist:
-        raise Exception("NFT substore entry does not exist")
-
-    for entry in getAttributesArray(nftID):
-        if entry['module'] == module:
-            return(entry['attributes'])
-
-    raise Exception("Specific module did not set any attributes.")
-```
-
-#### getLockingModule
-
-This function returns the locking status of an NFT.
-
-```python
-def getLockingModule(nftID: NFTID) -> Module:
-
-    if NFTStore[nftID] does not exist:
-        raise Exception("NFT substore entry does not exist")
-
-    if len(getNFTOwner(nftID)) == LENGTH_CHAIN_ID:
-        raise Exception("NFT is escrowed to another chain")
-    else:
-        return userStore[getNFTOwner(nftID), nftID].lockingModule
-```
-
-#### getNFTOwner
+#### getNFT
 
-This function returns the owner of an NFT.
+This function returns NFT and user substores information for a provided `nftID`.
 
 ```python
-def getNFTOwner(nftID: NFTID) -> Address | ChainID:
-
+def getNFT(nftID: NFTID) -> dict:
     if NFTStore[nftID] does not exist:
         raise Exception("NFT substore entry does not exist")
-
-    return NFTStore[nftID].owner
+    owner = NFTStore[nftID].owner    
+    attributesArray = [
+            {
+                "module": attribute.module,
+                "attributes": attribute.attributes
+            }
+            for attribute in NFTStore[nftID].attributesArray
+    ]
+    nftIDInfo = {
+        "owner": owner,
+        "attributesArray": attributesArray
+    }
+    if len(owner) == LENGTH_ADDRESS:
+        lockingModule = userStore[owner, nftID].lockingModule 
```

Comment:
To be consistent, I think it would be good to check if the user substore entry exists.

Reply:
Done in [this commit](https://github.com/LiskHQ/lips/pull/477/commits/876ae2c110612053e85a0c54fb65574a7e1b4bc1).